packages/eas-cli/src/commands/workflow/create.ts

```diff
@@ -0,0 +1,107 @@
+import chalk from 'chalk';
+import fs from 'fs/promises';
+import path from 'path';
+import prompts from 'prompts';
+
+import EasCommand from '../../commandUtils/EasCommand';
+import Log from '../../log';
+
+const WORKFLOWS_DIR = '.eas/workflows';
+const DEFAULT_WORKFLOW_NAME = 'workflow.yml';
+const HELLO_WORLD_TEMPLATE = `name: Hello World
+
+on:
+  push:
+    branches: ['*']
+
+jobs:
+  Hello World:
+    steps:
+      - run: echo "Hello, World"
+`;
+
+export class WorkflowCreate extends EasCommand {
+  static override description = 'create a new workflow configuration YAML file';
+
+  static override args = [
+    {
+      name: 'name',
+      description: 'Name of the workflow file (must end with .yml or .yaml)',
+      required: false,
+    },
+  ];
+
+  async runAsync(): Promise<void> {
+    const {
+      args: { name: argFileName },
+    } = await this.parse(WorkflowCreate);
+
+    let fileName = argFileName;
+
+    if (!fileName) {
+      const response = await prompts({
+        type: 'text',
+        name: 'fileName',
+        message: 'What would you like to name your workflow file?',
+        initial: DEFAULT_WORKFLOW_NAME,
+        validate: value => {
+          try {
+            this.validateYamlExtension(value);
+            return true;
+          } catch (error) {
+            return error instanceof Error ? error.message : 'Invalid file name';
+          }
+        },
+      });
+
+      if (!response.fileName) {
+        Log.warn('Workflow creation cancelled.');
+        process.exit(0);
+      }
+
+      fileName = response.fileName;
+    }
+
+    try {
+      await this.ensureWorkflowsDirectoryExistsAsync();
+      await this.createWorkflowFileAsync(fileName);
+      Log.withTick(`Created ${chalk.bold(path.join(WORKFLOWS_DIR, fileName))}`);
+    } catch (error) {
+      Log.error('Failed to create workflow file.');
+      throw error;
+    }
+  }
+
+  private validateYamlExtension(fileName: string): void {
+    const fileExtension = path.extname(fileName).toLowerCase();
+    if (fileExtension !== '.yml' && fileExtension !== '.yaml') {
+      throw new Error('File must have a .yml or .yaml extension');
+    }
+  }
```

Comment:
I believe we are doing the same thing in `workflow:validate`:
https://github.com/expo/eas-cli/blob/fdb167df344516d376b395242e5300b64fff8d5c/packages/eas-cli/src/commands/workflow/validate.ts#L55-L61

Let's extract it to the reusable helper function vs class method.